patches/minecraft/net/minecraft/entity/projectile/ProjectileHelper.java.patch

```diff
@@ -9,3 +9,12 @@
                    if (d0 == 0.0D) {
                       entity = entity1;
                       vec3d = vec3d1;
+@@ -156,7 +156,7 @@
+    public static Hand func_221274_a(LivingEntity p_221274_0_, Item p_221274_1_) {
+       return p_221274_0_.func_184614_ca().func_77973_b() == p_221274_1_ ? Hand.MAIN_HAND : Hand.OFF_HAND;
+    }
+-
++   public static Hand getHandWith(LivingEntity living, Predicate<Item> itemPredicate) { return itemPredicate.test(living.func_184614_ca().func_77973_b()) ? Hand.MAIN_HAND : Hand.OFF_HAND; }
```

Comment:
For both of these methods, 1) You can use multiple lines for full functions to make it not run off the edge.
2) Honestly this shouldn't be predicate. It should be getRangedWeapon or something like that that looks for bows/crossbows. So that all the callers don't have to specify the type every call.